### DISCUSSION TITLE
Multiple celery workers don't work using Kafka broker

### DISCUSSION THREAD

**Discussion by mfaw** [celery/kombu#1803] [Announcements]
I'm trying to have many celery workers running, multiple workers, not one worker with concurrency. I want to have multiple worker nodes.
I first tried using rabbitmq as a broker to celery and it worked with multiple workers and all workers sync together making tasks split among the open workers.
I tried doing the exact same thing with Kafka as a broker however, the workers don't sync and the tasks are sent to one worker only which is the last worker ready.
Any idea how I can fix this?
Thanks

**Comment by TayfurYldz** [comment]
The important Kafka concept here is the consumer group. Kombu's current `confluentkafka` transport creates a consumer with `group.id` defaulting to `<queue>-consumer-group` (and allows consumer configuration to override it). Kafka distributes a topic's partitions among consumers in the same group; it does not distribute one partition concurrently across several consumers.

So first check the number of partitions for the topic backing that Celery queue and confirm all workers are actually using the same queue name / `group.id`. If there is only one partition, only one worker in that consumer group can actively receive from it at a time. Increasing worker count without increasing partitions won't increase parallel consumption.

Also note that Kombu's Kafka transport is not behaviorally identical to AMQP/RabbitMQ; exchange/routing semantics and acknowledgement behavior are mapped onto Kafka primitives. I would validate the topology with a single queue, N partitions and N workers before carrying over a RabbitMQ deployment configuration unchanged.
